src/main_segment/calc.c

```diff
@@ -34,17 +104,79 @@ s16 angleF2S(f32 arg0) {
     return (s32)(f32)(arg0 * (0x10000 / 360.0));
 }
 
-INCLUDE_ASM("asm/nonmatchings/main_segment/calc", func_8007C4D8);
+/**
+ * Convert an Mtx to an MtxF.
+ */
+void func_8007C4D8(MtxF *mtxf, Mtx *mtx) {
+    s32 i;
+
+    for (i = 0; i < 4; i++) {
+        mtxf->mf[i][0] = (f32)mtx->m[i][0] / 0x8000;
+        mtxf->mf[i][1] = (f32)mtx->m[i][1] / 0x8000;
+        mtxf->mf[i][2] = (f32)mtx->m[i][2] / 0x8000;
+        mtxf->mf[i][3] = (f32)mtx->m[i][3] / 0x8000;
+    }
+}
+
+/**
+ * Convert an Mtx to an MtxF.
+ */
+void func_8007C540(Mtx *mtx, MtxF *mtxf) {
+    s32 i;
+
+    for (i = 0; i < 4; i++) {
+        mtx->m[i][0] = mtxf->mf[i][0] * 0x8000;
+        mtx->m[i][1] = mtxf->mf[i][1] * 0x8000;
+        mtx->m[i][2] = mtxf->mf[i][2] * 0x8000;
+        mtx->m[i][3] = mtxf->mf[i][3] * 0x8000;
+    }
+}
 
-INCLUDE_ASM("asm/nonmatchings/main_segment/calc", func_8007C540);
+/**
+ * Convert an Mtx to an MtxF, and set the last row to (0, 0, 0, 1).
+ */
+void func_8007C5A8(Mtx *mtx, MtxF *mtxf) {
+    s32 i;
 
-INCLUDE_ASM("asm/nonmatchings/main_segment/calc", func_8007C5A8);
+    for (i = 0; i < 3; i++) {
+        mtx->m[i][0] = mtxf->mf[i][0] * 0x8000;
+        mtx->m[i][1] = mtxf->mf[i][1] * 0x8000;
+        mtx->m[i][2] = mtxf->mf[i][2] * 0x8000;
+        mtx->m[i][3] = mtxf->mf[i][3] * 0x8000;
+    }
+    mtx->m[3][0] = 0;
+    mtx->m[3][1] = 0;
+    mtx->m[3][2] = 0;
+    mtx->m[3][3] = 1.0f * 0x8000;
+}
 
-INCLUDE_ASM("asm/nonmatchings/main_segment/calc", func_8007C624);
+/**
+ * Rotate `(pointX, pointY)` about `(centreX, centreY)` by `angle` degrees, modifying in-place.
+ */
+void func_8007C624(f32 angle, f32 centreX, f32 centreY, f32 *pointX, f32 *pointY) {
+    f32 angleRad = angle * (3.141592653589793 / 180.0);
```

Comment:
Does `M_PI` not match here?
Actually `gu.h` has `M_DTOR` which seems to be this same construction